src/Kiota.Builder/Extensions/OpenApiSchemaExtensions.cs

```diff
@@ -47,10 +47,19 @@ public static string GetSchemaName(this OpenApiSchema schema)
         return schema.GetSchemaNames().LastOrDefault()?.TrimStart('$') ?? string.Empty;// OData $ref
     }
 
+    public static OpenApiSchema? GetSchema(this OpenApiSchema schema)
+    {
+        if (schema is null)
+            return null;
+
+        return schema.IsArray() ? schema.Items : schema;
+    }
```

Comment:
Method name is generic, not very explicit. We do have a FlattenIfRequired method that already exists. I suggest you remove this new method and use that one instead where needed.